dbms/src/Server/Client.cpp

```diff
@@ -560,7 +561,22 @@ class Client : public Poco::Util::Application
                 const char * pos = begin;
                 ASTPtr ast = parseQuery(pos, end, true);
                 if (!ast)
+                {
+                    if (ignore_error)
+                    {
+                        // skip rest of bad query
+                        while (begin < end && *begin != ';')
```

Comment:
Trashy code.

Reply:
1. There is `find_first_symbols` and even `strchr` function.
2. The code doesn't do any lexical analysis and will fail with `ELECT ';'` query